src/main/java/uk/gov/companieshouse/officerfiling/api/interceptor/OfficersCRUDAuthenticationInterceptor.java

```diff
@@ -39,10 +59,37 @@ public boolean preHandle(@NonNull HttpServletRequest request,
                 (Map<String, String>) request.getAttribute(
                         HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
         final var transactionId = pathVariables.get(TRANSACTION_ID_KEY);
+        final String reqId = request.getHeader(ERIC_REQUEST_ID_KEY);
```

Comment:
Code seems reasonable, could this logic be reused by other CH services though? If so perhaps it could be a standalone interceptor in the security library? https://github.com/companieshouse/api-security-java/tree/main/src/main/java/uk/gov/companieshouse/api/interceptor

Reply:
Our OfficerCRUDAuthenticationInterceptor should really be merged with the library and there is an existing  comment to say that but I could not find a simpler way to add the company number check than to do it in our auth interceptor. Certainly something that should in future be standardised.